enhancements/authentication/direct-external-oidc-provider.md

```diff
@@ -0,0 +1,222 @@
+  - "@stlaz"
+  - "@ibihim"
+approvers:
+  - "@stlaz"
+  - "@deads2k"
+api-approvers:
+  - "@stlaz"
+creation-date: 2024-05-28
+last-updated: 2024-05-28
+tracking-link:
+  - https://issues.redhat.com/browse/OCPSTRAT-306
+see-also:
+  - "/enhancements/authentication/direct-oidc-study/study-oidc-in-openshift.md"
+replaces:
+  - ""
+superseded-by:
+  - ""
+---
+# Direct External OIDC Provider
+
+## Summary
+
+OpenShift has its own built-in OAuth server which can be used to obtain OAuth access tokens for authentication to the API. The server can be configured with an external identity provider (including support for OIDC), however it is still the built-in server that issues tokens. This enhancement proposal suggests a mechanism to enable configuration and direct usage of an external OIDC provider to issue tokens for authentication, instead of using the built-in OAuth server.
+
+## Motivation
+
+While external OIDC provider integration is supported by the built-in OAuth server, it is limited to the capabilities of the OAuth server itself. Customers want to be able to directly integrate their Identity Providers to the OpenShift cluster in order to facilitate machine-to-machine (e.g. CLI) workflows and capabilities of OIDC providers (similar to upstream Kubernetes), and to achieve seamless authentication in hybrid environments (e.g. k8s and non-k8s clusters) using a single Identity Provider.
+
+### User Stories
+
+- As a customer, I want to integrate my OIDC Identity Provider directly with OpenShift so that I can fully use its capabilities in machine-to-machine workflows.
+- As a customer in a hybrid cloud environment, I want to seamlessly use my OIDC Identity Provider across all of my fleet.
+
+### Goals
+
+1. Provide a direct authentication workflow such that OpenShift can consume bearer tokens issued by external OIDC identity providers.
+2. Replace the built-in OAuth stack by deactivating/removing its components as necessary.
+
+### Non-Goals
+
+n/a
+
+## Proposal
+
+This proposal introduces changes to the cluster-authentication-operator which manages the OAuth stack and to components that act as clients to the existing built-in OAuth server. The built-in OAuth server will still be available as the default option; the user will be able to configure their provider as a Day-2 configuration.
+
+Currently, any component that needs to obtain tokens or authenticate users does so against the built-in OAuth server. In order to integrate and use an external OIDC Identity Provider directly, any client component must replace its configuration to call the external OIDC provider instead of the built-in server. The components that act as OAuth server clients are:
+- OpenShift Console calls the OAuth server for user login to the console, and to obtain and display API access tokens
+- `oc` calls the OAuth server for user login to the API, and to obtain API access tokens
+- kube-apiserver calls the OAuth server via an authentication webhook for token management
```

Comment:
the enhancement statement is not true - the KAS reaches out to the oauth-apiserver